### PR TITLE
Use chromsizes from cooler files

### PR DESCRIPTION
**Pull request by pkerpedjiev**
## Description

This pull request allows using cooler files as chromosome size sources.

Why is it necessary?

To allow adding grid overlays without entering a chromsizes-tsv file.

## Checklist

- [x] Unit tests added or updated
- [x] Documentation added or updated
- [o] Example added or updated
- [o] Screenshot for visual changes (e.g. new tracks)
- [x] Updated CHANGELOG.md

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,8 @@
-## v1.3.1
+## v1.4.1
+
+- Add chromosome grid and label overlays directly from cooler files
```
Comment (flekschas) [inline CHANGELOG.md]: Didn't we add the chrom grid in v1.4?

File: app/scripts/configs/available-track-types.js [hunk 2]
```diff
@@ -21,20 +21,24 @@ import { DATATYPE_TO_TRACK_TYPE } from '.';
  */
 export const AVAILABLE_TRACK_TYPES = (datatypes, orientation) => {
   const datatypesToTrackTypes = DATATYPE_TO_TRACK_TYPE(orientation);
+  let availableTrackTypes = new Set(
+    Object.values(datatypesToTrackTypes)
+      .flatMap(x => x)
```
Comment (flekschas) [inline app/scripts/configs/available-track-types.js]: I've never heard of `flatMap`! Cool stuff

File: karma.conf.js [hunk 3]
```diff
@@ -21,7 +21,8 @@ module.exports = (config) => {
       'node_modules/bootstrap/dist/css/bootstrap.min.css',
       'node_modules/font-awesome/css/font-awesome.css',
       'build/hglib.css',
-      'test/**/*.+(js|jsx)',
+      // 'test/**/*.+(js|jsx)',
+      'test/AddTrackTests.js',
```
Comment (flekschas) [inline karma.conf.js]:
```suggestion
     //  'test/AddTrackTests.js',
```

File: karma.conf.js [hunk 4]
```diff
@@ -21,7 +21,8 @@ module.exports = (config) => {
       'node_modules/bootstrap/dist/css/bootstrap.min.css',
       'node_modules/font-awesome/css/font-awesome.css',
       'build/hglib.css',
-      'test/**/*.+(js|jsx)',
+      // 'test/**/*.+(js|jsx)',
```
Comment (flekschas) [inline karma.conf.js]:
```suggestion
      'test/**/*.+(js|jsx)',
```

File: CHANGELOG.md (hunk 1 above)
Comment (pkerpedjiev) [reply CHANGELOG.md]: What I meant to say is that chromsizes can be pulled directly from cooler files. This mostly just adds that as an option to the Add Track Dialog.

File: app/scripts/configs/available-track-types.js (hunk 2 above)
Comment (pkerpedjiev) [reply app/scripts/configs/available-track-types.js]: Oh yeah, once you know about flatMap it's hard to imagine living without it ;-)

File: CHANGELOG.md (hunk 1 above)
Comment (pkerpedjiev) [reply CHANGELOG.md]: I'll change the message.

**Pull request merged**